Update scatter_reduce_ test with parallel backend check

leslie-fang-intel opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #118708

**Summary**
Follow up of https://github.com/pytorch/pytorch/pull/118278, in which new added UT `test_scatter_using_atomic_add` failed with `native parallel backend` as reported in https://github.com/pytorch/pytorch/issues/118518.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

leslie-fang-intel committed:
Update scatter_reduce_ test with parallel backend check

[ghstack-poisoned]

leslie-fang-intel commented:
@pytorchbot merge

lezcano commented on test/inductor/test_cpu_repro.py:
```diff
@@ -1457,11 +1457,14 @@ def _internal_check(
-
-        with config.patch({"cpp.dynamic_threads": True}), set_num_threads(1):
-            _internal_check(fn, inps, "aten.scatter_reduce_")
+        if "ATen parallel backend: OpenMP" in torch.__config__.parallel_info():
```
May be better to check `torch.has_openmp`

leslie-fang-intel replied:
Thanks for the suggestions and changing to use `torch.has_openmp` is clearer.
Let's check if it can fix the issue. Since from the code, https://github.com/pytorch/pytorch/blob/b8307513e57f8beaf99daff342a23d705a417e11/caffe2/CMakeLists.txt#L16-L49, seems when `INTERN_BUILD_MOBILE` turns on, `AT_PARALLEL_NATIVE` and `USE_OPENMP` can be set to True at same time.

leslie-fang-intel committed:
Update on "Update scatter_reduce_ test with parallel backend check"


**Summary**
Follow up of https://github.com/pytorch/pytorch/pull/118278, in which new added UT `test_scatter_using_atomic_add` failed with `native parallel backend` as reported in https://github.com/pytorch/pytorch/issues/118518.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

leslie-fang-intel commented:
@pytorchbot merge

leslie-fang-intel commented:
@pytorchbot revert -m "Test Case still fail"

leslie-fang-intel commented:
@pytorchbot revert -m "Test Case still fail" -c nosignal

leslie-fang-intel committed:
Update on "Update scatter_reduce_ test with parallel backend check"


**Summary**
Follow up of https://github.com/pytorch/pytorch/pull/118278, in which new added UT `test_scatter_using_atomic_add` failed with `native parallel backend` as reported in https://github.com/pytorch/pytorch/issues/118518.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

leslie-fang-intel commented:
> May be better to check `torch.has_openmp`

Hi @lezcano, with checking `torch.has_openmp`, the test stills fails on `parallelnative-linux-jammy-py3.8-gcc11` which builds pytorch with `ATEN_THREADING=NATIVE`.
`USE_OPENMP` and `ATEN_THREADING` are 2 different building config as in `setup.py`. So, `torch.has_openmp` and using `native thread pool` can be `True` at the same time. To skip this test on `parallelnative-linux-jammy-py3.8-gcc11`, I am afraid we still need to check the msg from `torch.__config__.parallel_info()` for what's the `ATen parallel backend` is using,

lezcano reviewed (commented):
Ag, I see. Fair enough.
